### PR TITLE
Flashlight Package Manager plugin

### PR DESCRIPTION
**Pull request by mmarcon**
Flashlight Package Manager allows Flashlight users to install any plugin that is hosted on Github. In order to work with Flashlight Package Manager the plugin must be in its own repository and the root of the repository must contain the `info.json` file.

### CODE REVIEW

File: PluginDirectories/1/fpm.bundle/info.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "categories" : [
+    "Utilities",
+    "Package Management",
```
Comment (nate-parrott) [inline PluginDirectories/1/fpm.bundle/info.json]: Not sure I want to add two whole new top-level categories, but this doesn't really fit in the existing categories.

Would it be weird to have a "meta" category for flashlight-related stuff (e.g. the Flashlight plugin search plugin)?

File: PluginDirectories/1/fpm.bundle/README.markdown [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Flashlight Package Manager allows Flashlight users to install any plugin that is hosted on Github. In order to work with Flashlight Package Manager the plugin must be in its own repository and the root of the repository must contain the `info.json` file.
+
+It is good practice to add a `version` property to `info.json`. It is not used at the moment, it's just displayed in Spotlight, however versioning will be supported in the future.
+
+## Supported commands
+
+### Install plugins
+
+    fpm install ~id(a)
```
Comment (nate-parrott) [inline PluginDirectories/1/fpm.bundle/README.markdown]: I see that ~id can be any github path ($user/$project) — would it be possible to have the command accept full github URLs too? it'd be nice to be able to copy+paste from the browser.

**Comment by nate-parrott** [comment]
Wow, this is really impressive 😮. Thanks!

File: PluginDirectories/1/fpm.bundle/info.json (hunk 1 above)
Comment (mmarcon) [inline PluginDirectories/1/fpm.bundle/info.json]: That is really up to you. Maybe you could support the concept of tags along with categories and somehow enforce the restriction on what categories one can use in the categories array. I don't know. Having a model where any developer is free to make up their own categories but you would like to maintain a fixed list of categories is kind of hard to maintain. You'd have to basically add this very same comment to any pull request for plugins.

File: PluginDirectories/1/fpm.bundle/README.markdown (hunk 2 above)
Comment (mmarcon) [inline PluginDirectories/1/fpm.bundle/README.markdown]: Sure, I can do that. And I say that as I know how to do stuff in Python... but I am very good at Googling, I am sure I can figure it out. I can probably do it with a regular expression.

**Pull request merged**